main.py

```diff
@@ -27,17 +33,56 @@
             # get the full path to the file *after* the base reproschema_folder path
             # since files can be referenced by relative paths, we need to keep track of relative location
             filename = str(file.relative_to(reproschema_folder))
-            reproschema_content[filename] = load_file(file)
+            with open(f"{reproschema_folder}/{filename}") as f:
+                reproschema_content[filename] = json.loads(str(f.read()))
+
+    schema_name = [
+        name for name in list(reproschema_content.keys())
+        if name.endswith("_schema")
+    ][0]
+    reproschema_schema = reproschema_content[schema_name]
 
-    # TODO: validate reproschema
-    if reproschema_content["schema"] != "1.0.0-rc1":
-        raise ValueError('Unable to work with reproschema versions other than 1.0.0-rc1')
+    if (f"schema:version" in reproschema_schema and
+            reproschema_schema["schema:version"] not in ("0.0.1", "1.0.0-rc1")
+        ) or f"version" in reproschema_schema and reproschema_schema[
+            "version"] not in ("0.0.1", "1.0.0-rc1"):
+        raise ValueError(
+            'Unable to work with reproschema versions other than 0.0.1 or 1.0.0-rc1'
+        )
+    # before we print to file we wish to validate the jsons using fhir resources
 
     # convert to fhir
     config = Config()
-    questionnaire_generator = QuestionnaireGenerator(config=config)
-    fhir_questionnaire = questionnaire_generator.convert_to_fhir(reproschema_content)
 
+    questionnaire_generator = QuestionnaireGenerator(config)
+    fhir_questionnaire = questionnaire_generator.convert_to_fhir(
+        reproschema_content)
+
+    try:
+        questionnaire_json = construct_fhir_element('Questionnaire',
+                                                    fhir_questionnaire)
+    except:
+        raise ValueError(f"The Questionnaire json is not properly structured")
```

Comment:
I don't think these try except blocks are useful. They mask the real traceback, which is likely more informative. I would get rid of them.